install/uninstall.go

```diff
@@ -94,25 +99,25 @@ func (k *K8sUninstaller) Uninstall(ctx context.Context) error {
 		k.client.DeleteDaemonSet(ctx, k.params.Namespace, defaults.NodeInitDaemonSetName, metav1.DeleteOptions{})
 	}
 
-	if k.params.Wait {
-		k.Log("⌛ Waiting for Cilium to be uninstalled...")
+	k.Log("🔥 Deleting secret with the helm values configuration...")
+	k.client.DeleteSecret(ctx, k.params.Namespace, k.params.HelmValuesSecretName, metav1.DeleteOptions{})
+
+	k.Log("✅ Cilium was successfully uninstalled.")
+
+	return nil
+}
 
-	retry:
+func (k *K8sUninstaller) waitForPodsToBeDeleted(ctx context.Context) error {
+	for {
 		pods, err := k.client.ListPods(ctx, k.params.Namespace, metav1.ListOptions{LabelSelector: defaults.AgentPodSelector})
 		if err != nil {
 			return err
 		}
 
 		if len(pods.Items) > 0 {
```

Comment:
unrelated nit: if the context is cancelled while we're sleeping, we won't notice it. Can we fix that?